hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -2840,7 +2844,8 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
      * if the write-behind queue has reached its per-node maximum
      * capacity.
      */
-    Map<K, Object> executeOnEntries(EntryProcessor entryProcessor, Predicate predicate);
+    <R> Map<K, R> executeOnEntries(EntryProcessor<? super K, ? super V, R> entryProcessor,
+                                   Predicate<K, V> predicate);
```

Comment:
I think `? super K, ? super V` for `Predicate`?

Reply:
`Predicate` and `Projection` improvements are reserved for a separate PR to keep this one focused and to minimise the number of changed lines as it's already pretty big.